runtime/testy/studentAPI.py

```diff
@@ -134,19 +153,25 @@ def is_running(self, fn):
 
     return fn in self._coroutines_running
 
-  def _check_params(self, uid, param):
+  def _check_write_params(self, uid, param):
+    device_type = uid >> 72
+    valid_params = self.deviceName_to_writeParams[SENSOR_TYPE[device_type]]
+    if param not in valid_params:
+      raise StudentAPITypeError("Invalid param passed in, valid parameters for this device are: " + "".join(valid_params))
+
+  def _check_read_params(self, uid, param):
     device_type = uid >> 72
-    valid_params = self.deviceType_to_validParams[device_type]
+    valid_params = self.deviceName_to_readParams[SENSOR_TYPE[device_type]]
     if param not in valid_params:
       raise StudentAPITypeError("Invalid param passed in, valid parameters for this device are: " + "".join(valid_params))
 
   def _check_value(self, param, value):
     valid_values = self.param_to_valid_values[param]
     if not isinstance(value, valid_values[0]):
-      raise StudentAPIValueError("Invalid value type passed in, valid types for this param are: " + valid_values[0].__name__)
-    if valid_values[0] == bool:
+      raise StudentAPIValueError("Invalid value type passed in, valid types for this param are: " + valid_values[0][0].__name__)
```

Comment:
So technically each parameter has one valid type, its just that we need to count integers also as a float that we need the original list of both.